stac_fastapi/types/stac_fastapi/types/stac.py

```diff
@@ -83,3 +75,66 @@ class Collections(TypedDict, total=False):
 
     collections: List[Collection]
     links: List[Dict[str, Any]]
+
+
+class PartialCollection(TypedDict, total=False):
+    """Partial STAC Collection."""
+
+    type: Optional[str]
+    stac_version: Optional[str]
+    stac_extensions: Optional[List[str]]
+    id: Optional[str]
+    title: Optional[str]
+    description: Optional[str]
+    links: List[Dict[str, Any]]
+    keywords: Optional[List[str]]
+    license: Optional[str]
+    providers: Optional[List[Dict[str, Any]]]
+    extent: Optional[Dict[str, Any]]
+    summaries: Optional[Dict[str, Any]]
+    assets: Optional[Dict[str, Any]]
+
+
+class PartialItem(TypedDict, total=False):
+    """Partial STAC Item."""
+
+    type: Optional[Literal["Feature"]]
+    stac_version: Optional[str]
+    stac_extensions: Optional[List[str]]
+    id: Optional[str]
+    geometry: Optional[Dict[str, Any]]
+    bbox: Optional[BBox]
+    properties: Optional[Dict[str, Any]]
+    links: Optional[List[Dict[str, Any]]]
+    assets: Optional[Dict[str, Any]]
+    collection: Optional[str]
+
+
+class PatchAddReplaceTest(BaseModel):
+    """Add, Replace or Test Operation."""
+
+    path: str
+    op: Literal["add", "replace", "test"]
+    value: Any
+
+
+class PatchRemove(BaseModel):
+    """Remove Operation."""
+
+    path: str
+    op: Literal["remove"]
+
+
+class PatchMoveCopy(BaseModel):
+    """Move or Copy Operation."""
+
+    path: str
+    op: Literal["move", "copy"]
+
+    def __init__(self, *args, **kwargs):
+        """Init function to add 'from' field."""
+        super().__init__(*args, **kwargs)
+        self.__setattr__("from", kwargs["from"])
```

Comment:
because we're using pydantic model I think we can use Field

```python
class PatchMoveCopy(BaseModel):
   """Move or Copy Operation."""

   path: str
   op: Literal["move", "copy"]
   fromurl: str = Field(alias="from")
```

but then the attribute is `.fromurl` not `.from`

Reply:
Ah I hadn't seen `alias` before that looks neater.

I've also added:
```
model_config = ConfigDict(populate_by_name=True)
```
which means you can do this:
```
patch_operation = PatchMoveCopy(**{"path": "hello", "from":"world", "op": "move"})
```

And overrode the `model_dump` method so `by_alias` defaults to `True` to use the alias in the model dump 
```
patch_operation.model_dump()
#> {'path': 'hello', 'op': 'move', 'from': 'world'}
```
Not sure if this is the right thing to do or if we should expect people to use:
```
patch_operation.model_dump(by_alias=True)
```